omegatrade-app/backend/app/controller/simulation.controller.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+const Simulation = require('../models/simulation.model')
+const Company = require('../models/company.model')
+const { v4: uuidv4 } = require('uuid');
+const { spannerNumericRandVal, spannerNumericRandValBetween, generateRandomValue } = require('../helpers/stockdata.helper');
+const fakeStockmarketgenerator = require('fake-stock-market-generator');
+const { Spanner } = require('@google-cloud/spanner');
+
+
+/**
+ * Function to list all simulations.
+ * 
+ * @method GET
+ */
+exports.getList = async function (req, res) {
+    try {
+        await Simulation.getAll(function (err, data) {
+            if (err) {
+                res.json({ success: false, message: "something went wrong" });
+            }
+            if (data) {
+                res.status(200).json({ success: true, data: data });
+            }
+        });
+    } catch (error) {
+        return res.status(500).json({ success: false, message: "something went wrong" });
+    }
+};
+
+/**
+ * Function to update simulation
+ * 
+ * @method PUT
+ */
+exports.updateSimulation = async function (req, res) {
+    const body = req.body;
+    if (body) {
+        await Simulation.updateById(body, function (err, data) {
+            if (err) {
+                res.json({ success: false, message: "something went wrong" });
+            }
+            if (data) {
+                res.status(200).json({ success: true, message: `Simulation ${(body.status == true) ? 'Started' : 'Stopped'}  sucessfully` });
+            }
+        });
+    } else {
+        res.status(501).json({ success: false, message: "invalid data" });
+    }
+}
+
+/**
+ * Function to Delete simulation
+ * 
+ * @method DELETE
+ */
+exports.deleteSimulation = async function (req, res) {
+    let sId = req.params.sId;
+    if (sId) {
+        await Simulation.deleteById(sId, function (err, data) {
+            if (err) { res.json({ success: false, message: "something went wrong" }); }
+            if (data) { res.status(200).json({ success: true, message: `deleted sucessfully` }); }
+        });
+    } else {
+        res.status(501).json({ success: false, message: "Invalid data" });
+    }
+};
+
```

Comment:
@vignesh-sdev I see somewhere try catch has been added and somewhere its not added. any reason?

Reply:
No Specific Reason, Added try-catch as similar to other functions.